srml/democracy/src/lib.rs

```diff
@@ -492,6 +492,7 @@ mod tests {
 			transfer_fee: 0,
 			creation_fee: 0,
 			vesting: vec![],
+			_genesis_phantom_data: rstd::marker::PhantomData,
```

Comment:
This is pretty horrible...

Reply:
indeed,
When GenesisConfig needs to access to storage it needs to be generic other the trait. This trait is sometimes contained in GenesisConfig fields and sometimes not, in the latter cases we already add this field _genesis_phantom_data.

Now I changed the condition to requires phantom data also when the trait is generic over instances.
But this condition does actually have false positive. For instance the balances module could also not have a phantom, the Instance generic type is used in fields like T::Balance.
But as we don't access Trait definition inside the macro there is no way to know if T::Balance is actually an associated type from Trait (and so use the Instance generic type parameter) or an associated type from a dependency of Trait like T::AccountId would be (and so doesn't use Instance generic type parameter).

To summarize: This field is not always needed, it is not for balances, but there is no way to know inside the macro. The only solution I see is: we could provide a way for the user to say no_phantom_data (this seems good and would make balances module truly backward compatible)

Also note that user could write instead:
```rust
		t.extend(GenesisConfig::<Test>{
			dummy: 42,
			foo: 24,
			.. Default::default()
		}.build_storage().unwrap().0);
```